docs/env_var_reference.rst

```diff
@@ -0,0 +1,76 @@
+Environment Variables
+=====================
+CumulusCI has environment variables that are useful when CumulusCI is being run inside of web applications such as MetaCI, MetaDeploy, and Metecho.
+The following is a reference list of available environment variables that can be set.
+
+
+
+``CUMULUSCI_AUTO_DETECT`` 
+
+    Set this environment variable to autodetect branch and commit information from ``HEROKU_TEST_RUN_BRANCH`` and ``HEROKU_TEST_RUN_COMMIT_VERSION`` environment variables.
+
+
+
+``CUMULUSCI_DISABLE_REFRESH``
+
+    If present, will instruct CumulusCI to not refresh OAuth tokens for orgs.
+
+
+
+``CUMULUSCI_KEY``
+
+    An alphanumeric string used to encrypt org credentials at rest when an OS keychain is not available.
+
+
+
+``CUMULUSCI_REPO_URL``
+    
+    Used for specifying a Github Repository for CumulusCI to use when running in a CI environment.
+
+
+
+``GITHUB_APP_ID``
+
+    Your GitHub App's identifier.
+
+
+
+``GITHUB_APP_KEY``
+
+    Contents of a JSON Web Token (JWT) used to authenticate a GitHub app (see `GitHub’s docs <https://developer.github.com/apps/building-github-apps/authenticating-with-github-apps/#authenticating-as-a-github-app>`_).
+
+
+
+``GITHUB_TOKEN``
+
+    A GitHub `personal access token <https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line>`_.
+
+
+
+``HEROKU_TEST_RUN_BRANCH``
+
+    Used for specifying a specific branch to test against in a Heroku CI environment
+
+
+
+``HEROKU_TEST_RUN_COMMIT_VERSION``
+
+    Used to specify a specific commit to test against in a Heroku CI environment.
+
+
+
+``SFDX_CLIENT_ID``
+
+    Client ID for a Connected App used to authenticate to a persistent org, e.g. a Developer Hub. Set with SFDX_HUB_KEY.
+
+
+
+``SFDX_HUB_KEY``
+    
+    Contents of JSON Web Token (JWT) used to authenticate to a persistent org, e.g. a Developer Hub.  Set with SFDX_CLIENT_ID.
+
```

Comment:
We should note that when both SFDX_CLIENT_ID and SFDX_HUB_KEY are set, CumulusCI will use the JWT oauth flow to refresh org access tokens instead of the refresh token flow.